packages/core/lib/messages/baseMessageSchemas.ts

```diff
@@ -0,0 +1,22 @@
+import z from 'zod'
+
+export const BASE_MESSAGE_SCHEMA = z.object({
```

Comment:
@dariacm @CarlosGamero @Dosexe 
I've tried to revise the original structure that we used for defining our events previously to make it more straightforward, please review this change carefully. Migrating from one structure to another one is a major pain, let's ensure we've got this really right.